version: "3"
services:
  web:
    image: web-jupyterhub:latest
    environment:
      - CONFIGPROXY_AUTH_TOKEN=the-sign-pointed-this-way
      - ANNOUNCEMENT_JUPYTERHUB_API_TOKEN=the-hands-of-fate-have-doomed-this-man
    volumes:
      - ./config:/config
    ports:
      - 8081:8081
  app:
    image: app-notebooks:latest
    volumes:
      - ./config:/config
  announcement:
    image: web-announcement:latest
    environment:
      - JUPYTERHUB_API_TOKEN=the-hands-of-fate-have-doomed-this-man
      - JUPYTERHUB_API_URL=http://web:8081/hub/api
  proxy:
    image: jupyterhub/configurable-http-proxy:latest
    environment:
      - CONFIGPROXY_AUTH_TOKEN=the-sign-pointed-this-way
    ports:
      - 8000:8000
  service1:
    command: ["python", "-m", "http.server", "--directory", "/content/"]
    image: continuumio/miniconda3:latest
    ports:
      - 9000:8000
    volumes:
     - ./app-service1:/content
  service2:
    command: ["python", "-m", "http.server", "--directory", "/content/"]
    image: continuumio/miniconda3:latest
    ports:
      - 9001:8000
    volumes:
     - ./app-service2:/content
